companion.py

```diff
@@ -222,83 +250,85 @@ def authorize(self, payload):
                 tokens = tokens + [''] * (len(cmdrs) - len(tokens))
                 tokens[idx] = data.get('refresh_token', '')
                 config.set('fdev_apikeys', tokens)
-                config.save()	# Save settings now for use by command-line app
+                config.save()  # Save settings now for use by command-line app
+
                 return data.get('access_token')
+
         except:
-            print('Auth\tCan\'t get token for %s' % self.cmdr)
+            print('Auth\tCan\'t get token for {}'.format(self.cmdr))
             print_exc()
-            if r: self.dump(r)
-            raise CredentialsError()
+            if r:
+                self.dump(r)
 
-        print('Auth\tCan\'t get token for %s' % self.cmdr)
+            raise CredentialsError()  # TODO(A_D): Probably give more info about the error here
+
+        print('Auth\tCan\'t get token for {}'.format(self.cmdr))
         self.dump(r)
-        if data.get('error_description'):
-            raise CredentialsError('Error: %s' % data['error_description'])
-        elif data.get('error'):
-            raise CredentialsError('Error: %s' % data['error'])
-        elif data.get('message'):
-            raise CredentialsError('Error: %s' % data['message'])
-        else:
-            raise CredentialsError()
+        error = next((data[k] for k in ('error_description', 'error', 'message') if k in data), ('<unknown error>',))
+        raise CredentialsError('Error: {!r}'.format(error)[0])
 
     @staticmethod
     def invalidate(cmdr):
-        print('Auth\tInvalidated token for %s' % cmdr)
+        print('Auth\tInvalidated token for {}'.format(cmdr))
         cmdrs = config.get('cmdrs')
         idx = cmdrs.index(cmdr)
         tokens = config.get('fdev_apikeys') or []
         tokens = tokens + [''] * (len(cmdrs) - len(tokens))
         tokens[idx] = ''
         config.set('fdev_apikeys', tokens)
-        config.save()	# Save settings now for use by command-line app
+        config.save()  # Save settings now for use by command-line app
 
     def dump(self, r):
-        print('Auth\t' + r.url, r.status_code, r.reason and r.reason or 'None', r.text)
+        print('Auth\t' + r.url, r.status_code, r.reason if r.reason else 'None', r.text)
 
     def base64URLEncode(self, text):
         return base64.urlsafe_b64encode(text).decode().replace('=', '')
 
 
 class Session(object):
-
     STATE_INIT, STATE_AUTH, STATE_OK = list(range(3))
 
     def __init__(self):
         self.state = Session.STATE_INIT
         self.credentials = None
         self.session = None
         self.auth = None
-        self.retrying = False	# Avoid infinite loop when successful auth / unsuccessful query
+        self.retrying = False  # Avoid infinite loop when successful auth / unsuccessful query
 
         # yuck suppress InsecurePlatformWarning under Python < 2.7.9 which lacks SNI support
-        if sys.version_info < (2,7,9):
+        if sys.version_info < (2, 7, 9):
```

Comment:
This probably doesnt need to exist anymore, given that this file is not py2 compatible and the intent is to remove py2 support

Reply:
Yes, rip out anything that's specific to Python < 3.0.